### PR TITLE
feat:CXAR-918 Binding Agents Code

### PR DESCRIPTION
**Pull request by drcgjung**
## Description

Overtakes/Migrates Source Code from catenax-ng/product-agents
Fixes https://jira.catena-x.net/browse/KA-324

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up to date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] Copyright and license header are present on all affected files

### Additional information

- To request a review, check out [who's involved](https://projects.eclipse.org/projects/automotive.tractusx/who) to see a list of contributors and committers
- Use the [mailing list](https://accounts.eclipse.org/mailing-list/tractusx-dev) if you are unsure, who to contact, or if want to engage in a general discussion
- Check out our guide on [how to contribute](https://eclipse-tractusx.github.io/docs/oss/how-to-contribute)
- Check out our [Release Guidelines](https://eclipse-tractusx.github.io/docs/release)
- Check out the [Eclipse Foundation contribution guidelines](https://www.eclipse.org/projects/handbook/#contributing)

### CODE REVIEW

**Review by carslen — CHANGES_REQUESTED** [review]
Same issues as with PR #1 - License and Copyright header needs clarification. 

After clarifying this PR will be used for intital contribution to Eclipse-Tractusx and the required IP Check.

**Comment by drcgjung** [comment]
> Same issues as with PR #1 - License and Copyright header needs clarification.
> 
> After clarifying this PR will be used for intital contribution to Eclipse-Tractusx and the required IP Check.

maven/mavencentral/io.mikael/urlbuilder/2.0.9, , restricted, clearlydefined 
see https://github.com/mikaelhg/urlbuilder/blob/master/LICENSE

maven/mavencentral/io.swagger.core.v3/swagger-jaxrs2/2.2.12, , restricted, clearlydefined
see https://github.com/swagger-api/swagger-core/blob/master/LICENSE

maven/mavencentral/it.unibz.inf.ontop/ontop-model/5.0.2, , restricted, clearlydefined
maven/mavencentral/it.unibz.inf.ontop/ontop-obda-core/5.0.2, , restricted, clearlydefined
maven/mavencentral/it.unibz.inf.ontop/ontop-rdb/5.0.2, , restricted, clearlydefined

see https://github.com/ontop/ontop/blob/version5/LICENSE.txt

maven/mavencentral/org.apache.commons/commons-rdf-simple/0.5.0, , restricted, clearlydefined
see https://github.com/apache/commons-rdf/blob/master/LICENSE

maven/mavencentral/org.javabits.jgrapht/jgrapht-core/0.9.3, NOASSERTION, restricted, clearlydefined
see https://github.com/jgrapht/jgrapht/blob/master/license-EPL.txt

File: charts/remoting-agent/values.yaml
Comment (carslen) [inline charts/remoting-agent/values.yaml:28]: Do not use ghcr.io but Docker Hub. Please refer to [TRG 4.05 - Container registries](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-05).

File: charts/remoting-agent/values.yaml
Comment (carslen) [inline charts/remoting-agent/values.yaml:157]: Default value to ingress should be `enabled: false`. Please refer to [TRG 5.01](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-01). 

Also there should be no ingress-nginx related annotations as one could use a different ingress controller. Maybe just comment out and add an appropriate comment when to use these annotations.

File: conforming/README.md
Comment (carslen) [inline conforming/README.md:78]: Do not use ghcr.io but Docker Hub. Please refer to [TRG 4.05 - Container registries](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-05).

This comment addresses all other apprearances of ghcr.io as well.

File: conforming/README.md
Comment (carslen) [inline conforming/README.md:122]: After the initial contribution you should have GH workflows in place which will release your application and other artifacts like Helm chart.

Do not link to catenax-ng repositories/GH Org.

File: conforming/src/main/docker/Dockerfile
Comment (carslen) [inline conforming/src/main/docker/Dockerfile:27]: Use Eclipse Temurin as base image for Java based applications instead of alpine and adding additional packages to the image. Please refer to [TRG 4.02 - Base images](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-02).

File: provisioning/README.md
Comment (carslen) [inline provisioning/README.md:56]: Do not link to catenax-ng repositories/GH Org. After consortia phase has ended this might be inaccessible.

File: provisioning/README.md
Comment (carslen) [inline provisioning/README.md:127]: Do not use ghcr.io but Docker Hub. Please refer to [TRG 4.05 - Container registries](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-05).

This comment addresses all appearances of ghcr.io.

File: provisioning/README.md
Comment (carslen) [inline provisioning/README.md:277]: After the initial contribution you should have GH workflows in place which will release your application and other artifacts like Helm chart.

Do not link to catenax-ng repositories/GH Org.

File: provisioning/README.md
Comment (carslen) [inline provisioning/README.md:326]: These URL redirect service is pointing to catenax-ng. Do not link/reference to catenax-ng repositories or GH org.

File: remoting/README.md
Comment (carslen) [inline remoting/README.md:159]: Do not user ghcr.io but Docker Hub. This comment addresses all appearances of ghcr.io.

File: remoting/README.md
Comment (carslen) [inline remoting/README.md:204]: After the initial contribution you should have GH workflows in place which will release your application and other artifacts like Helm chart.

Do not link to catenax-ng repositories/GH Org.

File: remoting/src/main/java/org/eclipse/tractusx/agents/remoting/Application.java
Comment (carslen) [inline remoting/src/main/java/org/eclipse/tractusx/agents/remoting/Application.java:59]: This URL redirect service is pointing to catenax-ng GH org. Do not link/reference catenax-ng repositories and/or GH org.

This comment addresses all appearances of this redirect service.

File: remoting/src/test/resources/config.ttl
Comment (carslen) [inline remoting/src/test/resources/config.ttl:27]: URL redirect service pointing to catenax-ng. Do not link/reference catenax-ng repositories/GH org.

File: settings.xml
Comment (carslen) [inline settings.xml:32]: Although this is only a comment, this is not catenax-ng but Eclipse-Tractusx.

**Review by carslen — CHANGES_REQUESTED** [review]
I did not find a workflow for Helm Chart releasing? Did I miss it? Also you need to publish your docker container(s) to Docker Hub instead to ghcr.io (and do not use the images/packages build in catenax-ng!).

**Comment by carslen** [comment]
Opened IP-issues for restricted dependencies (#4) and updated `DEPENDENCIES` file with commit eca98d6

File: charts/conforming-agent/Chart.yaml
Comment (drcgjung) [reply charts/conforming-agent/Chart.yaml:28]: icon removed

File: charts/conforming-agent/README.md
Comment (drcgjung) [reply charts/conforming-agent/README.md:31]: switched to the eclipse tractusx helm repo (dev)

File: charts/conforming-agent/README.md.gotmpl
Comment (drcgjung) [reply charts/conforming-agent/README.md.gotmpl:33]: switched to the eclipse tractusx helm repo (dev)

File: charts/conforming-agent/values.yaml
Comment (drcgjung) [reply charts/conforming-agent/values.yaml:28]: switched to tractusx/conforming-agent (dockerhub)

File: charts/conforming-agent/values.yaml
Comment (drcgjung) [reply charts/conforming-agent/values.yaml:157]: done

File: charts/provisioning-agent/Chart.yaml
Comment (drcgjung) [reply charts/provisioning-agent/Chart.yaml:28]: icon removed

File: charts/provisioning-agent/README.md
Comment (drcgjung) [reply charts/provisioning-agent/README.md:31]: switched to eclipse tractusx helm repo (dev)

File: charts/provisioning-agent/README.md.gotmpl
Comment (drcgjung) [reply charts/provisioning-agent/README.md.gotmpl:33]: same here

File: charts/provisioning-agent/resources/cx-ontology.ttl
Comment (drcgjung) [reply charts/provisioning-agent/resources/cx-ontology.ttl:18]: done

File: charts/provisioning-agent/resources/portal.toml
Comment (drcgjung) [reply charts/provisioning-agent/resources/portal.toml:28]: done

File: charts/provisioning-agent/values.yaml
Comment (drcgjung) [reply charts/provisioning-agent/values.yaml:195]: ok

File: charts/provisioning-agent/README.md
Comment (drcgjung) [reply charts/provisioning-agent/README.md:31]: switched to the eclipse tractusx helm repo (dev)

File: charts/remoting-agent/Chart.yaml
Comment (drcgjung) [reply charts/remoting-agent/Chart.yaml:28]: icon removed

File: charts/remoting-agent/README.md.gotmpl
Comment (drcgjung) [reply charts/remoting-agent/README.md.gotmpl:32]: switched to eclipse tractusx helm repo (dev)

File: charts/remoting-agent/values.yaml
Comment (drcgjung) [reply charts/remoting-agent/values.yaml:157]: done

File: conforming/README.md
Comment (drcgjung) [reply conforming/README.md:122]: switched to the eclipse tractusx helm repo (dev)

File: conforming/src/main/docker/Dockerfile
Comment (drcgjung) [reply conforming/src/main/docker/Dockerfile:27]: switched to eclipse-temurin:17.0.6_10-jre-alpine

File: charts/provisioning-agent/resources/cx-ontology.xml
Comment (drcgjung) [reply charts/provisioning-agent/resources/cx-ontology.xml:22]: ok

File: charts/provisioning-agent/values.yaml
Comment (drcgjung) [reply charts/provisioning-agent/values.yaml:28]: moved to dockerhub/tractusx

File: charts/remoting-agent/values.yaml
Comment (drcgjung) [reply charts/remoting-agent/values.yaml:28]: moved to dockerhub/tractusx

File: conforming/README.md
Comment (drcgjung) [reply conforming/README.md:78]: moved to dockerhub/tractusx

File: provisioning/README.md
Comment (drcgjung) [reply provisioning/README.md:127]: moved to dockerhub/tractusx

File: provisioning/README.md
Comment (drcgjung) [reply provisioning/README.md:277]: moved to eclipse tractusx helm repo (dev)

File: provisioning/README.md
Comment (drcgjung) [reply provisioning/README.md:326]: The ontology will stay with the association. The redirect services will make sure that once the association decides about the final repo, we can simply move the redirection. Other projects (EDC) use w3id org without havong registered their namespace at all (which is semantically not correct!)

File: remoting/README.md
Comment (drcgjung) [reply remoting/README.md:159]: switched to dockerhub/tractusx

File: remoting/README.md
Comment (drcgjung) [reply remoting/README.md:204]: switched to eclipse tractusx helm repo (dev)

File: remoting/src/main/java/org/eclipse/tractusx/agents/remoting/Application.java
Comment (drcgjung) [reply remoting/src/main/java/org/eclipse/tractusx/agents/remoting/Application.java:59]: 1. Catenax will stay as an assocation
2. It will have control over particular standards, schemas and ontologies (which tractusx may use in order to provide implementations according to these standards, schemas and ontologies).
3. Using a redirection service to make the actual location of these standards, schemas and ontologies is best practice.

File: remoting/src/test/resources/config.ttl
Comment (drcgjung) [reply remoting/src/test/resources/config.ttl:27]: will not comply here

**Comment by drcgjung** [comment]
> I did not find a workflow for Helm Chart releasing? Did I miss it? Also you need to publish your docker container(s) to Docker Hub instead to ghcr.io (and do not use the images/packages build in catenax-ng!).

workflows for chart linting and releasing have been added (see https://github.com/eclipse-tractusx/bpdm/blob/main/.github/workflows/helm-chart-lint.yaml and https://github.com/eclipse-tractusx/bpdm/blob/main/.github/workflows/helm-chart-release.yaml

For reference to the yet not published Agents Kit we chose an intermediate redirection https://bit.ly/tractusx-agents which will be moved to eclipse-tractusx with Release 3.3 

For referring the Catena-X Ontology (and Domain Ontologies, and Use Case Ontologies) which will be hosted by the Catena-X Association (in whatever repository, then) we use the https://w3id.org/catenax Namespace (see what EDC did without yet properly applying for their namespace!!!!) - this will point wherever the respective ontologies will be released.

If that is a problem, we need to escalate.
CGJ

**Comment by Siegfriedk** [comment]
Initial contribution ticket created on EF: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/9835

**Comment by Siegfriedk** [comment]
This one is still not done: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/9816 we need to wait for this to be able to merge it. 

Initial contribution is done; As soon as the last dependency is verified, pls regenerate the DEPENDENCIES file, we will then merge it and do everything else in the QG4 review

**Comment by drcgjung** [comment]
[INFO] Querying Eclipse Foundation for license data for 225 items.
[INFO] Found 191 items.
[INFO] Querying ClearlyDefined for license data for 34 items.
[INFO] Found 34 items.
[INFO] Vetted license information was found for all content. No further investigation is required.

**Comment by Siegfriedk** [comment]
@drcgjung merging it now & @carslen this is NOT an accepted PR for QG4 its only for preparing for QG4. we will do the full QG4 check after merging this.

**Pull request merged**
